data/pigui/modules/map-sector-view.lua

```diff
@@ -0,0 +1,229 @@
+local Engine = import('Engine')
+local Game = import('Game')
+local ui = import('pigui/pigui.lua')
+local Vector = import('Vector')
+local Color = import('Color')
+local Lang = import("Lang")
+local lc = Lang.GetResource("core");
+local lui = Lang.GetResource("ui-core");
+local utils = import("utils")
+local Event = import("Event")
+
+local player = nil
+local colors = ui.theme.colors
+local icons = ui.theme.icons
+
+local mainButtonSize = Vector(32,32) * (ui.screenHeight / 1200)
+local mainButtonFramePadding = 3
+local function mainMenuButton(icon, selected, tooltip, color)
+	if color == nil then
+		color = colors.white
+	end
+	return ui.coloredSelectedIconButton(icon, mainButtonSize, selected, mainButtonFramePadding, colors.buttonBlue, color, tooltip)
+end
+
+local function showSystemInfo(label, current_systempath, systempath, othersystempath)
+	if systempath then
+		local starsystem = systempath:GetStarSystem()
+		local clicked = false
+		ui.withID(label, function()
+								local jumpData = ""
+								if not current_systempath:IsSameSystem(systempath) then
+									local jumpStatus, distance, fuelRequired, duration = player:GetHyperspaceDetails(current_systempath, systempath)
```

Comment:
Note for the future : it might be time to revisit the API here. I put it together in haste back in the day, but I find the multiple return value dissatisfying.